Pod/Classes/Math/FloatingPoint+Scale.swift

```diff
@@ -13,11 +13,12 @@ public extension FloatingPoint {
     /// - Parameters:
     ///   - source: The range to interpret the number as being a part of.
     ///   - destination: The range to map the number to.
+    ///   - timingTransform: An optional transform mapping input percentage to output percentage. Defaults to `nil`
     ///   - clamped: Whether the result should be clamped to the `to` range. Defaults to `false`.
     ///   - reversed: whether the output mapping should be revserd, such that
     ///               as the input increases, the output decreases. Defaults to `false`.
     /// - Returns: The input number, scaled from the `from` range to the `to` range.
-    public func scaled(from source: ClosedRange<Self>, to destination: ClosedRange<Self>, clamped: Bool = false, reversed: Bool = false) -> Self {
+    public func scaled(from source: ClosedRange<Self>, to destination: ClosedRange<Self>, timingTransform: ((Self) -> Self)? = nil, clamped: Bool = false, reversed: Bool = false) -> Self {
```

Comment:
#nth For naming I would drop `timing`, since this function isn't inherently related to animation. Alternatives include `transform`, `function`, `mapFunction` etc. But actually this whole function is becoming a monster IMO, and the name `scaled` itself no longer represents what this function does

Reply:
Was chatting with @heyltsjay about this, and the method is a bit of a monster, but I'm not sure it's too much yet. It would be possible to model these options to try and reign this in, not sure if it helps things tho, ie:

```    enum Options {
        case clamped
        case reversed
        case curve(CurveProvider)
    }
func scaled(from source: ClosedRange<Self>, to destination: ClosedRange<Self>, options: [Option]) -> Self
```

Reply:
Can you do an OptionSet with associated values? If not, this looks like a nice approach.